var/spack/repos/builtin/packages/intel-mkl/package.py

```diff
@@ -142,6 +142,8 @@ def install(self, spec, prefix):
     def setup_dependent_environment(self, spack_env, run_env, dependent_spec):
         # set up MKLROOT for everyone using MKL package
         spack_env.set('MKLROOT', self.prefix)
+        spack_env.append_path('SPACK_COMPILER_EXTRA_RPATHS',
```

Comment:
i think ideally compiler wrappers should work more closely with `LibraryList` and alike which know everything about where each library lives, as opposed to assuming it's located in `prefix/lib` or `prefix/lib64`.